src/physics/grid/XsGridInterface.hh

```diff
@@ -38,12 +36,30 @@ struct XsGridPointers
     //! Whether the interface is initialized and valid
     explicit CELER_FUNCTION operator bool() const
     {
-        return log_energy && !value.empty()
+        return log_energy && (value.size() >= 2)
                && (prime_index < log_energy.size
                    || prime_index == size_type(-1))
                && log_energy.size == value.size();
     }
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * A generic grid of 1D data with arbitrary interpolation.
+ */
+struct GenericGridPointers
+{
+    Span<const real_type> grid;         //!< X coordinate
+    Span<const real_type> value;        //!< Y coordinate
```

Comment:
Is the X,Y nomenclature (in the comments) intended here? (Sound odd to have Y coordinate for 1D data :) ).

Reply:
I guess I was thinking in terms of the time-honored `y = f(x)` -- I can update the comment for clarity.

Reply:
Improved comments in #129

Reply:
thanks